src/axolotl/utils/callbacks.py

```diff
@@ -313,3 +315,155 @@ def on_evaluate(
+            #     eval_dataloader,
+            #     description="Evaluation",
+            #     # No point gathering the predictions if there are no metrics, otherwise we defer to
+            #     # self.args.prediction_loss_only
+            #     # prediction_loss_only=True if trainer.compute_metrics is None else None,
+            #     prediction_loss_only=False,
+            #     # ignore_keys=ignore_keys,
+            #     # metric_key_prefix=metric_key_prefix,
+            # )
+
+            # print(type(output))
+            # print(dir(output))
+            # print(output.predictions)
+            # print(output.label_ids)
+            # print(output.metrics)
+
+            # # Extract the predictions and labels from the output
+            # predictions = output.predictions
+            # labels = output.label_ids
+            # # Convert the predictions and labels to a readable format
+            # predictions = [tokenizer.decode(p) for p in predictions]
+            # labels = [tokenizer.decode(l) for l in labels]
+
+            # # Add the data to the wandb.Table
+            # for prediction, label in zip(predictions, labels):
+            #     table.add_data(prediction, label)
+
+            trainer.model.eval()
+            # preds, refs = [], []
+            # loss_bench = 0
+            predictions = []
+            for batch in tqdm(eval_dataloader, total=len(eval_dataloader)):
+                (loss, logits, labels) = trainer.prediction_step(
+                    trainer.model,
+                    batch,
+                    prediction_loss_only=False,
+                )
+
+                print("logits", logits)
+                print("labels", labels)
+
+                pred_tokens = []
+                for i, logit in enumerate(logits):
+                    print(dir(logit))
+                    print(logit)
+                    print(logit.shape)
+                    # # Convert the logits to probabilities using softmax
+                    # probabilities = torch.softmax(logit, dim=-1)
+
+                    # # Get the predicted token id (the one with the highest probability)
+                    # predicted_token_id = torch.argmax(probabilities).item()
+
+                    # # Decode the predicted token id to get the plaintext
+                    # predicted_token = tokenizer.decode([predicted_token_id])
+
+                    # # Append the predicted token to the preds list
+                    # pred_tokens.append(predicted_token)
+
+                    # Convert the logits to probabilities using softmax
+                    probabilities = torch.softmax(logit, dim=-1)
+
+                    # Get the predicted token ids (the ones with the highest probability)
+                    predicted_token_ids = torch.argmax(probabilities, dim=-1)
+
+                    # Decode the predicted token ids to get the plaintext
+                    predicted_tokens = tokenizer.batch_decode(predicted_token_ids)
```

Comment:
@Glavin001 create a little helper function for all of this manual decoding work with Pytorch & the tokenizer. Code should be much more readable / maintainable afterwards.